.github/workflows/beam_PreCommit_Python_ML.yml

```diff
@@ -104,8 +104,8 @@ jobs:
           arguments: |
             -Pposargs="${{
               contains(matrix.os, 'self-hosted') &&
-                'apache_beam/ml/ -m ''not require_non_self_hosted''' ||
-                'apache_beam/ml/ -m ''require_non_self_hosted'''
+                'apache_beam/ml/ -m ''not require_docker_in_docker''' ||
+                'apache_beam/ml/ -m ''require_docker_in_docker'''
```

Comment:
Hm, it looks like this will currently not work because of https://github.com/apache/beam/blob/8930c8dd5f8ef97cdceddbec792aec08455f56f7/sdks/python/scripts/run_pytest.sh#L34

Reply:
https://github.com/apache/beam/actions/runs/16300739721/job/46034393155

Reply:
This is the context - https://issues.apache.org/jira/browse/BEAM-12985

Reply:
Probably a simple fix would be to replace https://github.com/apache/beam/blob/8930c8dd5f8ef97cdceddbec792aec08455f56f7/sdks/python/scripts/run_pytest.sh#L34

and instead of failing, collect the extra `-m` arguments and add the appropriate `not no_xdist` and `no_xdist` flags in 

https://github.com/apache/beam/blob/8930c8dd5f8ef97cdceddbec792aec08455f56f7/sdks/python/scripts/run_pytest.sh#L44-L48